### PR TITLE
py-emodel-generalisation: allow latest libsonata 0.1.26

### PR DESCRIPTION
**Pull request by tristan0x**
This change updates the dependencies of spack package py-emodel-generalisation according to the corresponding Python package. Plus, it adds a couple of new package versions for py-bluecellulab.

This allows py-emodel-generalisation spack package to depend on the most recent version of libsonata 0.1.26.

### CODE REVIEW

File: bluebrain/repo-bluebrain/packages/py-emodel-generalisation/package.py [hunk 1]
```diff
@@ -23,19 +23,19 @@ class PyEmodelGeneralisation(PythonPackage):
...
+    depends_on("py-pyyaml@6:", type=("build", "run"))
     depends_on("py-bluepyopt@1.13.86:", type=("build", "run"))
     depends_on("py-neurom@3.2.2:", type=("build", "run"))
-    depends_on("py-efel@5.5.7:", type=("build", "run"))
-    depends_on("py-morph-tool@2.9:", type=("build", "run"))
+    depends_on("py-efel@5.5.5:", type=("build", "run"))
+    depends_on("py-morph-tool@2.9.1:", type=("build", "run"))
     depends_on("neuron+python@8.0:", type=("build", "run"))
     depends_on("py-matplotlib@3.6.2:", type=("build", "run"))
-    depends_on("py-bluecellulab@1.7.6:2.3.1", type=("build", "run"))
```
Comment (matz-e) [inline bluebrain/repo-bluebrain/packages/py-emodel-generalisation/package.py:33]:
```
emodel-generalisation 0.2.10 has requirement bluecellulab<=2.3.1,>=1.7.6, but you have bluecellulab 2.6.11.
```
?

**Review by matz-e — COMMENTED** [review]
Looked good, but there's still a version conflict

File: bluebrain/repo-bluebrain/packages/py-emodel-generalisation/package.py (hunk 1 above)
Comment (tristan0x) [reply bluebrain/repo-bluebrain/packages/py-emodel-generalisation/package.py:33]: WIP: https://bbpteam.epfl.ch/project/issues/browse/BBPP40-488

Comment (tristan0x) [reply bluebrain/repo-bluebrain/packages/py-emodel-generalisation/package.py:33]: @arnaudon  needs a few days before releasing a new version of py-emodel-generalisation. Until then, this draft PR will remain like this.

**Comment by arnaudon** [comment]
I'm not sure why it cannot find the new version, maybe I was to quick in updating things here, but I cannot retrigger

File: bluebrain/repo-bluebrain/packages/py-emodel-generalisation/package.py [hunk 2]
```diff
@@ -13,7 +13,7 @@ class PyEmodelGeneralisation(PythonPackage):
     git = "https://github.com/BlueBrain/emodel-generalisation.git"
     pypi = "emodel-generalisation/emodel-generalisation-0.2.5.tar.gz"
```
Comment (matz-e) [inline bluebrain/repo-bluebrain/packages/py-emodel-generalisation/package.py]: Naming is inconsistent, that is why it is not finding the files, /me thinks:
image

Comment (arnaudon) [reply bluebrain/repo-bluebrain/packages/py-emodel-generalisation/package.py]: ok, weird it worked before, I modified the dash to underscore

**Review by matz-e — APPROVED** [review]

**Pull request merged**